### DISCUSSION TITLE
Auth using Microsoft Entra ID

### DISCUSSION THREAD

**Discussion by Stadly** [langfuse/langfuse#1820] [Ideas]
### Describe the feature or potential improvement

Azure AD is [deprecated in Auth.js](https://authjs.dev/getting-started/providers/azure-ad). Would be nice to be able to use the env vars for [Microsoft Entra ID](https://authjs.dev/getting-started/providers/microsoft-entra-id) instead.

**Comment by marcklingen** [comment]
Got started working on this, but support for Entra is not released yet by Auth.js: https://authjs.dev/getting-started/providers/microsoft-entra-id

PR: https://github.com/langfuse/langfuse/pull/1822

**Reply by Stadly** [reply]
That's surprising. I updated Auth.js and got depracation warning for Azure AD: https://github.com/nextauthjs/next-auth/blob/9bb56dd5055321e0ffe1f8124eaef69ef97710d0/packages/core/src/providers/azure-ad.ts#L22

Maybe it's released for Auth.js but not for NextAuth.js?

**Reply by marcklingen** [reply]
Open for a PR on this if you can figure it out. Doesn’t seem to be super straightforward right now until this is released by nextauth/authjs

**Reply by Stadly** [reply]
Let's wait for the release :)

**Comment by marcklingen** [comment]
Update on this, EntraId works with the existing AzureAD provider. I have updated the docs accordingly.
